### ISSUE TITLE
Quickbooks integration

### ISSUE THREAD

**Issue by daN4cat** [opensourcepos/opensourcepos#2466]
Here in the UK the use of an approved accountancy electronic tool became mandatory, and you are supposed to keep electronic track of all the daily transactions, invoices and etc.
It's part of the digital economy initiative to transform the UK into a more digital country.

My wife's accountant uses Quickbooks and I need to find a means to push the end of the day transaction automatically so I don't need to do that silly job manually every week.

Quickbooks PHP SDK: https://github.com/intuit/QuickBooks-V3-PHP-SDK

**Comment by odiea** [comment]
Looks like a great addition for good old OSPOS. Thanks

**Comment by WebShells** [comment]
Other option is front accounting, doesn't have the modernity of quickbooks, yet rich in features if you'd like to have a look at...

**Comment by daN4cat** [comment]
The problem is that we don't have an "end of the day big button".
It's either I implement that or I implement a overnight autosync.

**Comment by daN4cat** [comment]
@WebShells HMRC here in the UK has a list of recognised tools and certified accountants want a SaaS solution that does for them as for the small businesses they look after.
To be honest I feel accountants are gaining out, because they are pushing all the data insert to small businesses and they just consolidate, yet asking the same money as before + QBO monthly fee!!!

**Comment by odiea** [comment]
What would be the idea here, Create a new beginning and End of day for Cashups? That would work. Also when the Register is first Entered maybe a Modal to add the beginning of Day cash in Register.

**Comment by daN4cat** [comment]
Maybe a Lock of the Cashup at the end of the day could do.

At the moment the workflow is you create a Cashup in the morning, and in the evening you reopen and submit with all the daily transaction + expenses data pre-filled in as check.
Maybe I could add a checkbox "lock" on the top to say this is final and on that trigger push the data to QBO.

**Comment by odiea** [comment]
Good Idea. I know when my wife had her thrift store she liked looking at the day totals through that or a similar feature the program had that she was using.

**Comment by WebShells** [comment]
That will replace end of day process, will still need beginning cash amount available in the register at the start of the day/shift and a reconcile function i think...

**Comment by odiea** [comment]
Also what I implemented was the cash in/out was done thru the expenses module and taken out of the opening cash drawer and not the total amount at the end of day. I made a few of the inputs read only so OSPOS would fill out the info and the employee did not have to.

**Comment by odiea** [comment]
But I think the Employee should enter the end of day cash in the drawer because money could disappear otherwise and OSPOS would show the true cash total that was not available to the employee. Just the way we had to do business here at the store she had.

**Comment by WebShells** [comment]
Yes, that's what i meant in reconcile process, it should show short or over amount so owner/manager can tell if cash or total transactions is equivalent to what's in the register...

**Comment by odiea** [comment]
Here is what I came up with. The just looking button is there so the current info is not saved after the initial open cash input. The permission is for someone who has permission to config. The columns on the right will not be seen by a normal user. Try it out if you want to. 4 parts controller,model,view and additional cashups tables. The in/out cash is entered in the expenses module for better tracking.
 
[cashups info.zip](https://github.com/opensourcepos/opensourcepos/files/3294271/cashups.info.zip)

![2019-06-16 09_42_48-Window](https://user-images.githubusercontent.com/10962177/59566327-f9bde800-901b-11e9-9585-e17411a8330f.png)

**Comment by daN4cat** [comment]
Going back to QBO, probably the easiest solution is to POST a SaleReceipt to QBO when creating a Cashup with 0 amount and then update it with PUTs when the subsequents submits are done.
Last Cashup submission of the day will be the last total of the SaleReceipt.

**Comment by WebShells** [comment]
@odeia to be honest i didnt do the right follow up for the cashup module since it was missing some of the required input, yet I'd like to check it out to see the results being calculated, just waiting for my eyes to be good back so i can resume 🙄

**Comment by andriux1990** [comment]
I loved this change!
is there any progress here?

**Comment by daN4cat** [comment]
@andriux1990 little progress I'm afraid.

**Comment by andriux1990** [comment]
Because we don't push this. I'm going to test thoroughly, day by day to see how it goes
